### PR TITLE
egress: update comments for types and add field validation

### PR DESCRIPTION
**Pull request by shashankram**
**Description**:
Updates the comments for the API types and adds validation
for the IPAddresses and Port fields.

Part of #3045

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`No`

### CODE REVIEW

File: charts/osm/crds/policy.yaml [hunk 1]
```diff
@@ -85,6 +86,7 @@ spec:
                       number:
                         description: Port number of this port.
                         type: integer
+                        pattern: ^[1-9]\d*$
```
Comment (eduser25) [inline charts/osm/crds/policy.yaml:89]: would be good to prevent non-uint16 ranges. guess this could also be done by validating webhook

**Review by eduser25 — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

File: charts/osm/crds/policy.yaml [hunk 2]
```diff
@@ -73,6 +73,7 @@ spec:
                   type: array
                   items:
                     type: string
+                    pattern: ((?:\d{1,3}.){3}\d{1,3})\/(\d{1,2})$
```
Comment (jaellio) [inline charts/osm/crds/policy.yaml:76]: Should the "." be escaped?

**Review by jaellio — APPROVED** [review]

File: charts/osm/crds/policy.yaml (hunk 1 above)
Comment (shashankram) [reply charts/osm/crds/policy.yaml]: Done!

File: charts/osm/crds/policy.yaml (hunk 2 above)
Comment (shashankram) [reply charts/osm/crds/policy.yaml:76]: It should, good catch!

**Pull request merged**